File: objective-c/ArcGISSample/Swipe/SwipeViewController.m
//
//  SwipeViewController.m
//  ArcGISSample
//
//  Created by esrij on 2015/07/27.
//  Copyright (c) 2015年 esrij. All rights reserved.
//

#import "SwipeViewController.h"


@implementation SwipeViewController



- (void)viewDidLoad
{
    [super viewDidLoad];
    
    //道路地図レイヤー表示用のマップの作成
    AGSMapView *agsMapView1 = [[AGSMapView alloc] initWithFrame:self.view.bounds];
	[self.view addSubview:agsMapView1];
    
    //マップにタイルマップサービスレイヤーを追加（道路地図）
	NSURL *url1 = [NSURL URLWithString:@"https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer"];
	AGSTiledMapServiceLayer *tiledMapServiceLayer1 = [[AGSTiledMapServiceLayer alloc] initWithURL:url1];
	[agsMapView1 addMapLayer:tiledMapServiceLayer1 withName:@"Tiled Layer1"];
    
    AGSEnvelope *envelope = [AGSEnvelope envelopeWithXmin:1.5557970122810215E7 ymin:4258398.013496462 xmax:1.5558175713936899E7  ymax:4258509.895960432  spatialReference:agsMapView1.spatialReference];
    [agsMapView1 zoomToEnvelope:envelope animated:NO];
    
    
    //衛星画像レイヤー表示用のUIViewの作成
    self.layerView = [[UIView alloc] initWithFrame:self.view.bounds];
    [self.view addSubview:self.layerView];
    
    //UIViewに衛星画像レイヤー表示用のマップを追加
    AGSMapView *agsMapView2 = [[AGSMapView alloc] initWithFrame:self.view.bounds];
	[self.layerView addSubview:agsMapView2];
    
    //マップにタイルマップサービスレイヤーを追加（衛星画像）
    NSURL *url2 = [NSURL URLWithString:@"https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer"];
	AGSTiledMapServiceLayer *tiledMapServiceLayer2 = [[AGSTiledMapServiceLayer alloc] initWithURL:url2];
	[agsMapView2 addMapLayer:tiledMapServiceLayer2 withName:@"Tiled Layer2"];
    [agsMapView2 zoomToEnvelope:envelope animated:NO];
    
    self.layerView.clipsToBounds = YES;
    
    UISlider *slider = [[UISlider alloc] initWithFrame:CGRectMake(0, 100, self.view.frame.size.width, 50)];
    slider.minimumValue = 0.0f;
    slider.maximumValue = 1.0f;
    [slider addTarget:self action:@selector(sliderEvent:) forControlEvents:UIControlEventValueChanged];
    [self.view addSubview:slider];
    
}

- (void)sliderEvent:(UISlider *)slider {
    
    //衛星画像レイヤー表示用のUIViewのframeを変更する
    [self.layerView setFrame:CGRectMake(0, 0, self.view.frame.size.width * slider.value, self.view.frame.size.height)];
    
}


- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: swift/ArcGISSample/SwipeViewController.swift
//
//  SwipeViewController.swift
//  ArcGISSample
//
//  Created by esrij on 2015/07/27.
//  Copyright (c) 2015年 esrij. All rights reserved.
//


import UIKit
import ArcGIS


class SwipeViewController: UIViewController{
    
    
    var layerView: UIView!
    
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        // 道路地図レイヤー表示用のマップを作成
        let mapView1 = AGSMapView(frame: view.bounds)
        view.addSubview(mapView1)
        let map1 = AGSMap(basemapType: AGSBasemapType.streets, latitude: 35.658581, longitude: 139.745433, levelOfDetail: 15)
        mapView1.map = map1
        
        // マップ操作を無効にする
        mapView1.isUserInteractionEnabled = false

        
        // 衛星画像表示用のマップを追加するUIViewを作成
        layerView = UIView(frame: view.bounds)
        view.addSubview(layerView)
        
        // 衛星画像レイヤー表示用のマップを作成
        let mapView2 = AGSMapView(frame: view.bounds)
        let map2 = AGSMap(basemapType: AGSBasemapType.imagery, latitude: 35.658581, longitude: 139.745433, levelOfDetail: 15)
        mapView2.map = map2
        mapView2.isUserInteractionEnabled = false

        // 衛星画像表示用のマップをUIViewに追加
        layerView.addSubview(mapView2)


        layerView.clipsToBounds = true
        
        // スライダーの作成
        let slider = UISlider(frame: CGRect(x: 0, y: 100, width: view.frame.size.width, height: 50))
        slider.minimumValue = 0.0
        slider.maximumValue = 1.0
        slider.addTarget(self, action: #selector(SwipeViewController.sliderEvent(sender:)), for: .valueChanged)
        view.addSubview(slider)
        
    }
    
    
    @objc func sliderEvent(sender: UISlider) {
        
        // 衛星画像レイヤー表示用のUIViewのframeを変更する
        layerView.frame = CGRect(x: 0, y: 0, width: view.frame.size.width * CGFloat(sender.value), height: view.frame.size.height)
    }
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
}
